Formula/lua.rb

```diff
@@ -42,6 +33,15 @@ class Lua < Formula
     end
   end
 
+  # Be sure to build a dylib, or else runtime modules will pull in another static copy of liblua = crashy
+  # See: https://github.com/Homebrew/legacy-homebrew/pull/5043
+  patch do
+    on_macos do
+      url "https://raw.githubusercontent.com/Homebrew/formula-patches/11c8360432f471f74a9b2d76e012e3b36f30b871/lua/lua-dylib.patch"
+      sha256 "a39e2ae1066f680e5c8bf1749fe09b0e33a0215c31972b133a73d43b00bf29dc"
+    end
+  end
```

Comment:
Can we move the Linux patch into this block?

Reply:
Oh yeah, good idea. I already found another thing to fix, so I'll do another run-through to find these things.

Reply:
@carlocab, do you think we should just require all `on_{system}` blocks to be nested inside `resource` blocks, even when there are multiple?

Currently, cases like `kpcli` allow this:

```ruby
on_linux do
  resource "foo"
    ...
  end
  resource "bar"
    ...
  end
end
resource "baz"
  ...
end
```

Instead, we could require:
```ruby
resource "foo"
  on_linux do
    ...
  end
resource "bar"
  on_linux do
    ...
  end
end
resource "baz"
  ...
end
```

Reply:
Also, with the `lua` example, we could combine them but I'm not sure we should since the patches are for different issues

Reply:
Why do you say that? The macOS and Linux patches are both for building a shared `liblua`.